File: callback.py
def process_callbacks(callbacks):
    for callback in callbacks:
        callback()

# Example usage
def callback1():
    print("Callback 1")

def callback2():
    print("Callback 2")

# Create a list of 100,000 callbacks
callbacks = [callback1] * 50000 + [callback2] * 50000

# Process the callbacks
process_callbacks(callbacks)

File: callback.go
package main

import "fmt"

type Callback func()

func ProcessCallbacks(callbacks []Callback) {
	for _, callback := range callbacks {
		callback()
	}
}

// Example callback functions
func Callback1() {
	fmt.Println("Callback 1")
}

func Callback2() {
	fmt.Println("Callback 2")
}

func main() {
	// Create a slice of 100,000 callbacks
	callbacks := make([]Callback, 100000)
	for i := range callbacks {
		if i%2 == 0 {
			callbacks[i] = Callback1
		} else {
			callbacks[i] = Callback2
		}
	}

	// Process the callbacks
	ProcessCallbacks(callbacks)
}
